rcd.go

```diff
@@ -0,0 +1,144 @@
+package factom
+
+import (
+	"crypto/ed25519"
+	"encoding"
+	"fmt"
+
+	"github.com/Factom-Asset-Tokens/factom/varintf"
+)
+
+// RCD is the underlying structure behind a factoid address. A factoid address is a sha256d(RCD).
+// The most common and basic RCD type, is type 1. That being just a single public key that uses a single
+// 64 byte signature block.
+type RCD interface {
```

Comment:
I can get on board with this. I really wanted to keep the ability to handle multiple RCDs, but that is a long ways out. The library will fail when a new RCD is released in my current version, so I can live with just assuming everything is RCD 1, and failing otherwise in the current release.

Reply:
Yeah lets keep things simple and specific until new RCD types are defined. then add the concrete implementations and then retro fit code that uses RCDs with the proper and limited interface abstraction.

The other thing to consider is that new RCD types may require a blockheight activation, which add some complexity to very generic implementations where that would not get caught.